elevation_mapping_costmap_2d_plugin/src/elevation_layer.cpp

```diff
@@ -0,0 +1,240 @@
+      if (hasEdgesLayer) {
+        const grid_map::Matrix& edgesData = elevationMap_[edgesLayerName_];
+        if (edgesData(gridmapIndex(0), gridmapIndex(1)) < edgesSharpnessThreshold_)  // If area not sharp, dont label as obstacle.
+        {
+          setCost(mx, my, FREE_SPACE);
+          continue;
+        }
+      }
+      setCost(mx, my, LETHAL_OBSTACLE);
+    } else {
+      setCost(mx, my, FREE_SPACE);
+    }
+  }
+
+  if (footprintClearingEnabled_) {
+    setConvexPolygonCost(transformedFootprint_, costmap_2d::FREE_SPACE);
+  }
+
+  switch (combinationMethod_) {
+    case Overwrite:
+      updateWithOverwrite(master_grid, min_i, min_j, max_i, max_j);
+      break;
+    case Maximum:
+      updateWithMax(master_grid, min_i, min_j, max_i, max_j);
+      break;
+    default:  // Do Nothing.
+      break;
+  }
+}
+
+CombinationMethod convertCombinationMethod(const std::string& str) {
+  if (boost::iequals(str, "Overwrite")) {  // Case insensitive comparison.
+    return Overwrite;
+  } else if (boost::iequals(str, "Maximum")) {  // Case insensitive comparison.
+    return Maximum;
+  } else {
+    ROS_WARN_THROTTLE(0.5, "Unknow combination method !");
+    return Unknown;
+  }
+}
+
+void ElevationLayer::elevationMapCallback(const grid_map_msgs::GridMapConstPtr& elevation) {
+  grid_map::GridMap incomingMap;
+  grid_map::GridMap filteredMap;
+  if (!grid_map::GridMapRosConverter::fromMessage(*elevation, incomingMap)) {
+    ROS_WARN_THROTTLE(0.5, "Grid Map msg Conversion failed !");
+    return;
+  }
+  lastElevationMapUpdate_ = ros::Time::now();
+  incomingMap.convertToDefaultStartIndex();
+  if (!(globalFrame_ == incomingMap.getFrameId())) {
+    ROS_WARN_THROTTLE(0.5, "Incoming elevation_map frame different than expected! ");
+  }
+  // Apply filter chain.
+  if (filtersConfigurationLoaded_ && filterChain_.update(incomingMap, filteredMap)) {
+    std::lock_guard<std::mutex> lock(elevationMapMutex_);
+    elevationMap_ = filteredMap;
+    heightThreshold_ /= 2.0;  // Half the threshold since the highest sharpness is at midheigth of the obstacles.
+  } else {
+    std::lock_guard<std::mutex> lock(elevationMapMutex_);
+    ROS_WARN_THROTTLE(0.5, "Could not use the filter chain!");
+    elevationMap_ = incomingMap;
+  }
+  if (!elevationMapReceived_) {
+    elevationMapReceived_ = true;
+  }
+}
+
+void ElevationLayer::setupDynamicReconfigure(ros::NodeHandle& nodeHandle_) {
+  dsrv_.reset(new dynamic_reconfigure::Server<elevation_mapping_costmap_2d_plugin::ElevationPluginConfig>(nodeHandle_));
+  dynamic_reconfigure::Server<elevation_mapping_costmap_2d_plugin::ElevationPluginConfig>::CallbackType cb =
+      boost::bind(&ElevationLayer::reconfigureCB, this, _1, _2);
+  dsrv_->setCallback(cb);
+}
+
+void ElevationLayer::reconfigureCB(elevation_mapping_costmap_2d_plugin::ElevationPluginConfig& config, uint32_t level) {
+  enabled_ = config.enabled;
+}
+
+void ElevationLayer::reset() {
+  deactivate();
+  resetMaps();
+  current_ = true;
```

Comment:
Current what? Please use clearer variable names.

Reply:
All those variables are derived from base classes in costmap_2d .....